ext/dcos-installer/dcos_installer/action_lib/__init__.py

```diff
@@ -16,6 +16,35 @@ class ExecuteException(Exception):
     """Raised when execution fails"""
 
 
+def get_successful_nodes_from(step, state_json_dir):
+    assert isinstance(step, str)
+    assert step in ['preflight', 'deploy', 'postflight']
+    assert state_json_dir is not None
+    assert isinstance(state_json_dir, str)
+
+    step += '.json'
+
+    step_path = os.path.join(state_json_dir, step)
+    step_json = _read_state_file(step_path)
+    if not step_json:
+        # TODO: Fix exception here
+        raise Exception("Could not get hosts from {}".format(step_path))
+
+    assert 'hosts' in step_json, "Field `hosts` not found, location {}".format(step_path)
+
+    nodes = []
+    for node_ip, node_params in step_json['hosts'].items():
+        assert 'host_status' in node_params, "Field `host_status` not found for node {}".format(node_ip)
+        if node_params['host_status'] == 'success':
+            assert 'tags' in node_params, 'Field `tags` not found for node {}'.format(node_ip)
+            nodes.append(Node(node_ip, tags=node_params['tags']))
+        else:
+            log.debug("node {} unhealthy {}".format(node_ip, node_params['host_status']))
+
+    assert nodes, "No healthy nodes found"
```

Comment:
having no successful nodes going from one stage to the next seems like it shouldn't be an assertion error, but is rather a runtime error / we need to fail in the UI but the code isn't wrong / incorrect.